model/src/model.py

```diff
@@ -129,37 +281,52 @@ def __init__(self, settings: Settings) -> None:
     def process_data_online(self, train_loader: DataLoader, test_loader: DataLoader) -> None:
         for epoch in range(self.settings.epochs):
             for i, batch in enumerate(train_loader):
+                # permute to (num_steps, batch_size, data_size)
                 batch = batch.permute(1, 0, 2)
-                batch = batch[:, :, :1]
 
-                print(
-                    f"Epoch {epoch} - Batch {i} - Sample data: {batch.shape}")
+                # poisson encode
+                spike_trains = spikegen.rate(batch, time_var_input=True)
```

Comment:
might not be needed for the initial two cluster data set.

Reply:
I agree, however it is a bit smelly that in that case S_j would be a continuous value. I was thinking it wasn't that big of a deal, however offline I thought of how S_j can actually be a continuous value above the spike upper limit of 1. This seems bad enough to warrant always Poisson encoding?